src/improvements/AddressRegistry.sol

```diff
@@ -410,34 +410,30 @@ contract AddressRegistry is IAddressRegistry, Test {
 
     function getAnchorStateRegistryProxy(address faultDisputeGame, address permissionedDisputeGame)
         internal
-        view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).anchorStateRegistry() returns (address anchorStateRegistryProxy) {
-            return anchorStateRegistryProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).anchorStateRegistry();
-        }
+        (bool ok, bytes memory data) =
+            address(faultDisputeGame).call(abi.encodeWithSelector(IFetcher.anchorStateRegistry.selector));
+        if (ok && data.length == 32) return abi.decode(data, (address));
+
+        return IFetcher(permissionedDisputeGame).anchorStateRegistry();
     }
 
     function getDelayedWETHProxy(address faultDisputeGame, address permissionedDisputeGame)
         internal
-        view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).weth() returns (address delayedWethProxy) {
-            return delayedWethProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).weth();
-        }
+        (bool ok, bytes memory data) = address(faultDisputeGame).call(abi.encodeWithSelector(IFetcher.weth.selector));
```

Comment:
We can change these all to staticcalls and keep the `view` modifier
```suggestion
        (bool ok, bytes memory data) = address(faultDisputeGame).staticcall(abi.encodeWithSelector(IFetcher.weth.selector));
```

Reply:
https://github.com/ethereum-optimism/superchain-ops/pull/582/commits/9fea16d01342df5fb6b77b9218b62376ebd61b2c